src/compiler/checker.ts

```diff
@@ -3651,9 +3651,7 @@ module ts {
                     var maybeCache = maybeStack[depth];
                     // If result is definitely true, copy assumptions to global cache, else copy to next level up
                     var destinationCache = result === Ternary.True || depth === 0 ? relation : maybeStack[depth - 1];
-                    for (var p in maybeCache) {
-                        destinationCache[p] = maybeCache[p];
-                    }
+                    copyMap(maybeCache, destinationCache);

```

Comment:
can you use named parameter comments to specify whichis the source, and which is the destination?  Thanks!